### ISSUE TITLE
Improve Safari async error message

### ISSUE THREAD

**Issue by henryiii** [pyodide/pyodide#2376]
When using the async runPython, the error message in Safari is very poor.

Here's a comparison of Safari and Chrome, Async:
<img width="1533" alt="Screen Shot 2022-04-08 at 4 50 41 PM" src="https://user-images.githubusercontent.com/4616906/162528477-f836cfcc-3d6c-480e-bafe-cb3292ea5dc3.png">

And Sync:
<img width="1533" alt="Screen Shot 2022-04-08 at 4 51 11 PM" src="https://user-images.githubusercontent.com/4616906/162528485-ecfdd66c-c16e-4ec4-a097-5f4e1002520a.png">

The actual error cause is not included in Safari. From #2373.

**Comment by hoodmane** [comment]
If you catch the error and look at `error.message`, does it include the whole text?

**Comment by henryiii** [comment]
Yes, it does. I'm just surrounding it in `try { ... } catch (error) { console.error(error); }` which does print the whole error. And there's a ... there that doesn't expand when you expand it, maybe there's some other way to expand it that I'm not aware of...

**Comment by hoodmane** [comment]
Hmm. Well it would fix the problem if you put `KeyError: 'tool'` first.

**Comment by henryiii** [comment]
I think the "normal" order is what Python users are used to, it's just Safari's fault for hard truncating it. It doesn't truncate it with `console.error`, though, and I can't see how to get at the whole error.

**Comment by henryiii** [comment]
Seems like Safari is bad about truncating strings. A warning/workaround (like surrounding the code in the try/catch) might be FAQ worthy, otherwise, feel free to close.

**Comment by hoodmane** [comment]
If you wouldn't mind adding an FAQ entry with this, that'd be helpful.
